portfolio/src/main/webapp/script.js

```diff
@@ -50,16 +50,28 @@ function showPics() {
 }
 
 /* Gets array list of messages from servlet, converts to JSON, and displays it on home page. */
-function getJSONMessages() {
+function getBucketList() {
   fetch('/data').then(response => response.json()).then((list) => {
     const listElement = document.getElementById('list-container');
     listElement.innerHTML = '';
     let bucketList = "";
 
-    for (let i = 0; i < list.length; i++) {
-      bucketList += (i + 1) + ": " + list[i] + '\n';
+    // Display elements in a numerical list
+    let count = 1;
+    for (i = 0; i < list.length; i++) {
+      if (list[i] === "") {
+        bucketList += "";
```

Comment:
No need to append an empty string. This could be a `continue;` statement, or the logic could be flipped so that you're checking 
```
if (list[i]) {
  // Do stuff
}
// Implicitly does nothing when the String is empty because this is the end of the loop body.
```